### ISSUE TITLE
Backport conformance fix for Listener condition reason ListenerReasonRefNotPermitted

### ISSUE THREAD

**Issue by sunjayBhatia** [kubernetes-sigs/gateway-api#1328]
**What would you like to be added**:
Backport changes from https://github.com/kubernetes-sigs/gateway-api/pull/1305 to release 0.5.x

**Why this is needed**:
Implementations that are updated to pass conformance tests on `main` now fail on the Gateway API 0.5.0 conformance tests (and vice versa). Since #1305 is a bug fix It would be great to ensure the latest release line is patched along with `main`

If this is an acceptable change, can start the process of making a PR to backport the change, will need some guidance on what to do if there is not yet a process around patch releases

(This in particular applies to Contour's implementation, see: https://github.com/projectcontour/contour/pull/4664)

**Comment by robscott** [comment]
Good point. I think we should release a v0.5.1 with the updates to our conformance tests that have merged recently.

**Comment by shaneutt** [comment]
I'm on board with a `v0.5.1` for the recent conformance test fixes :+1:

**Comment by sunjayBhatia** [comment]
I don't see a 0.5.x release branch yet, I assume that needs to be created before we can do a backport? (sidebar, maybe we need a separate issue to document the backport process for maintainers and contributors if it doesnt exist already?)

**Comment by shaneutt** [comment]
I've created `release-0.5` (analogous to `release-0.4`) :+1:

**Comment by youngnick** [comment]
I'm +1 on a patch release as well.

**Comment by sunjayBhatia** [comment]
submitted #1351 to backport the commit

**Comment by robscott** [comment]
Thanks @sunjayBhatia!
